scipy/optimize/_optimize.py

```diff
@@ -887,12 +887,12 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
         warnflag = 1
         msg = _status_message['maxfev']
         if disp:
-            print('Warning: ' + msg)
+            warnings.warn(msg, UserWarning)
```

Comment:
We should also add some `stacklevel` (2 or 3 usually depending on where it is). Cf https://github.com/scipy/scipy/issues/1953#issuecomment-954829478

Reply:
+1. Good call. I always forget about that.

Reply:
Same! I understood this one quite recently.

Reply:
I added.